File: cases/taint/kotlin/return-relay-one-hop-positive/ReturnRelayOneHopPositive.kt
package dataflowbench

object ReturnRelayOneHopPositive {
    fun dfb_source(): Int { // DFB-SOURCE: return-one-hop-input
        return 1
    }

    fun relay(value: Int): Int { // DFB-WITNESS: return-one-hop-relay
        return value
    }

    fun dfb_sink(value: Int) {} // DFB-SINK: return-one-hop-sink

    fun run() {
        val result = relay(dfb_source())
        dfb_sink(result)
    }
}

File: cases/taint/go/return-relay-one-hop-positive/return_relay_one_hop_positive.go
package dataflowbench

func dfb_source() int { // DFB-SOURCE: return-one-hop-input
	return 1
}

func relay(value int) int { // DFB-WITNESS: return-one-hop-relay
	return value
}

func dfb_sink(value int) {} // DFB-SINK: return-one-hop-sink

func run() {
	result := relay(dfb_source())
	dfb_sink(result)
}
